File: tourneyAPI/Models/Entities/Characters/DarkSamus.cs


namespace Entities;

using Enums;

// Defines the competitive profile metadata for this playable character.
public class DarkSamus : Character
{
    // Initializes this character's default competitive attributes for matchmaking and tier logic.
    public DarkSamus()
    {
        Id = CharacterId.DarkSamus;
        characterName = CharacterName.DARK_SAMUS;
        archetype = Archetype.TURTLE;
        fallSpeed = FallSpeed.FLOATY;
        weightClass = WeightClass.MIDDLEWEIGHT;
        tierPlacement = TierPlacement.A;
    }

}

File: tourneyClient/src/models/entities/Characters/DarkSamus.ts
import { CharacterName } from '@/models/Enums/CharacterName';
import { Character } from '../Character';
import { CharacterId } from '@/models/Enums/CharacterId';
import { Archetype } from '@/models/Enums/Archetype';
import { FallSpeed } from '@/models/Enums/FallSpeed';
import { WeightClass } from '@/models/Enums/WeightClass';
import { TierPlacement } from '@/models/Enums/TierPlacement';

export const DarkSamus: Character =
{
    id: CharacterId.DarkSamus,
    characterName: CharacterName.DARK_SAMUS,
    archetype: Archetype.TURTLE,
    fallSpeed: FallSpeed.FLOATY,
    weightClass: WeightClass.MIDDLEWEIGHT,
    tierPlacement: TierPlacement.A
}

export default DarkSamus; 